app/controllers/users_controller.rb

```diff
@@ -22,9 +22,47 @@ def create
     end
   end
 
+  def request_reset
+  end
+
+  def send_reset_link
+    user = User.where(email: params[:email]).first
+    if user
+      scheme = request.env['rack.url_scheme']
+      host = request.env['HTTP_HOST']
+      link = user.generate_reset_link(scheme, host)
+      AppMailer.send_password_reset_link(user, link).deliver
+      render 'confirm_password_reset'
+    else
+      flash[:danger] = 'If you forgot your email address as well, you can register again.'
+      render 'request_reset'
+    end
+  end
+
+  def enter_new_password

```

Comment:
In my opinion I feel like this is moving too far away from the standard CRUD model. I would suggest moving to a `passwords_controller` where you can use the standard CRUD method.